drivers/SmartThings/zigbee-fan/src/kichler-fan-light/init.lua

```diff
@@ -0,0 +1,179 @@
+  end
+end
+
+-- ZIGBEE HANDLERS
+local function zb_fan_control_handler(driver, device, value, zb_rx)
+  if value.value < 4 then
+    device:emit_event(capabilities.fanSpeed.fanSpeed(value.value))
+    local evt = capabilities.switch.switch(value.value > 0 and 'on' or 'off', { visibility = { displayed = true } })
+    device:emit_component_event(device.profile.components.main, evt)
+    device:emit_component_event(device.profile.components.main, capabilities.fanSpeed.fanSpeed(value.value))
+    if value.value > 0 then
+      device:set_field('LAST_FAN_SPD', value.value, {persist = true})
+    end
+  end
+end
+
+local function zb_level_handler(driver, device, value, zb_rx)
+  local evt = capabilities.switchLevel.level(math.floor(0.5 + (value.value / 254.0) * 100))
+  device:emit_component_event(device.profile.components.light, evt)
+end
+
+local function zb_onoff_handler(driver, device, value, zb_rx)
+  local attr = capabilities.switch.switch
+  local evt = value.value and attr.on() or attr.off()
+  device:emit_component_event(device.profile.components.light, evt)
+end
+
+local function info_changed(driver, device, event, args)
+  if device.preferences ~= nil then
+    local current_level = device:get_latest_state('light', capabilities.switchLevel.ID, capabilities.switchLevel.level.NAME) or 100
+    local new_trim = tonumber(device.preferences.trim)
+    local old_trim = tonumber(args.old_st_store.preferences.trim)
+    if args.old_st_store.preferences.trim ~= device.preferences.trim then
+      local newlevel = math.floor((current_level/100.0) * 254 )
+      if old_trim < new_trim then
+        if new_trim >= current_level then
+          newlevel = math.floor((new_trim/100.0) * 254 )
+        end
+      end
+      device:send_to_component('light', Level.server.commands.MoveToLevelWithOnOff(device, newlevel, 0))
+    end
+    if device.preferences.breezemode ~= args.old_st_store.preferences.breezemode then
+      local speed = device:get_field('LAST_FAN_SPD') or 1
+      local breeze_flag = tonumber(device.preferences.breezemode)
+      if breeze_flag == 0 then
+        device:send(FanControl.attributes.FanMode:write(device, 5))
+        device:send(FanControl.attributes.FanMode:write(device, speed))
+      elseif breeze_flag == 1 then
+        device:send(FanControl.attributes.FanMode:write(device, 4))
+      end
+    end
+    if device.preferences.fandirection ~= args.old_st_store.preferences.fandirection then
+      local flag_direction = device:get_field('FANDIRECTION_FLAG') or 0
+      local send_fandirection_time = device:get_field('FANDIRECTION_SENDTIME') or 0
+      local current_fandirection_time = os.time()
+      local time_difference = os.difftime(current_fandirection_time, send_fandirection_time) or 0
+      if time_difference >= 10 or send_fandirection_time == 0 then
+        device:send(FanControl.attributes.FanMode:write(device, 6))
+        device:set_field('FANDIRECTION_SENDTIME', current_fandirection_time, {persist = true})
```

Comment:
this value doesn't need to persist

it takes more than 10s for the hub to reboot

Reply:
The purpose of using the following code, regardless of the hub, is to prevent sending commands if a command has been sent within 10 seconds. (This is related to protecting the fan driving circuit.)

Reply:
I understand, but the `persist = true` makes the value durable through hub restarts.

Because hub restarts take more than 10s, you do not need to have this flag persist, because it is guaranteed to have been more than 10s since you last sent the write on hub restart. Just keeping it in memory should be enough.